src/image/hfe.c

```diff
@@ -375,11 +380,66 @@ static bool_t hfe_write_track(struct image *im)
             + (im->cur_track & 1) * 256
             + batch_off - im->hfe.write_batch.off
             + (off & 255);
-        for (i = 0; i < nr; i++)
+
+        i = 0;
+
+        if (is_v3 && off == im->hfe.write.start && off > 2) {
+            /* Avoid starting write in the middle of an opcode. */
+            if (*(w-2) == OP_SkipBits) {
+                w++;
+                i++;
+            } else {
+                switch (*(w-1)) {
+                case OP_SkipBits:
+                    w += 2;
+                    i += 2;
+                    break;
+                case OP_Bitrate:
+                    w++;
+                    i++;
+                    break;
+                default:
+                    break;
+                }
+            }
+        }
+
+        for (; i < nr; i++) {
```

Comment:
Proposal: Make this a `while` loop. Push updates of `i` into the loop (`w[i++]` and so on). It's actually harder work to read this having to account for the update to `i` in the loop header.